huaweicloud/services/elb/data_source_huaweicloud_elb_listeners.go

```diff
@@ -0,0 +1,282 @@
+			},
+			"loadbalancer_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"protocol": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"protocol_port": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"enterprise_project_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"listener_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"listeners": {
+				Type:     schema.TypeList,
+				Elem:     listenersSchema(),
+				Computed: true,
+			},
+		},
+	}
+}
+
+func listenersSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"protocol": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"protocol_port": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"default_pool_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"http2_enable": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"forward_eip": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"forward_port": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"forward_request_port": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"forward_host": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"sni_certificate": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"server_certificate": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ca_certificate": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"tls_ciphers_policy": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"idle_timeout": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"request_timeout": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"response_timeout": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"loadbalancer_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"advanced_forwarding_enabled": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"protection_status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"protection_reason": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+	return &sc
+}
+
+func dataSourceElbListenersRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	var (
+		listLoadListenersHttpUrl = "v3/{project_id}/elb/listeners"
```

Comment:
```suggestion
		listListenersHttpUrl = "v3/{project_id}/elb/listeners"
```
and fix other places